kyo-prelude/shared/src/main/scala/kyo2/Abort.scala

```diff
@@ -58,21 +57,25 @@ object Abort:
         def apply[A, S, ES, ER](v: => A < (Abort[E | ER] & S))(
             using
             ct: ClassTag[E],
+            tag: Tag[E], // TODO Used only to ensure E isn't a type union. There should be a more lightweight solution for this.
```

Comment:
@hearnadam I'm using `Tag` here only as a way to deny the use of type unions for `E`. We should have a proper solution for this as we discussed in private.

Reply:
We can explore a more unified solution for this sort of problem once intersection/union types are supported by a single String tag.